### ISSUE TITLE
Ubuntu Virtual Machine setup issue

### ISSUE THREAD

**Issue by nbibin001** [Parrot-Developers/RollingSpiderEdu#49]
Hi I was trying to try out some new controls with the Rolling Spider with Matlab Simulink. i dont have Ubuntu installed in a machine. So I decided to use the virtual machine as per the instructions. I installed VMWare and was setting the virtual machine using the 1630FCS_VM virtual disk file. I am attaching my screenshots, it initially shows a linux selection screen and I selected Ubuntu from the list. Then it shows me a message saying that the CPU has been disabled by guest operating system and shows a black screen as shown. Could you please let me know if I am missing something ? Appreciate your help.

![obuntu_vm_11](https://cloud.githubusercontent.com/assets/15659590/18695688/89a03e7c-7f81-11e6-96ca-ce4a0decf4d5.PNG)
![obuntu_vm_12](https://cloud.githubusercontent.com/assets/15659590/18695698/97d76e20-7f81-11e6-9649-c6e4d92b89da.PNG)
![obuntu_vm_13](https://cloud.githubusercontent.com/assets/15659590/18695703/9d5064ce-7f81-11e6-93c0-238f883a7d49.PNG)

**Comment by friether** [comment]
Looks like a generic problem with getting a VM up and running. were you able to create and run a virtual OS generally?

**Comment by nbibin001** [comment]
Hi thanks a lot for the reply. I haven't tried any other virtual OS other than the Obuntu virtual disk file shared for this example. Any suggestions how I can get around ? Thanks for your help again.

**Comment by hborhan** [comment]
Hi, I have similar problem. For my case, I installed VMware 12.5 and tried to connect to the virtual machine but it gives an error. Is there any specific version of VMware we need to have to connect? Was anyone here successful recently to use the VM files on the MIT website?

**Comment by guivenca** [comment]
Hey,
I was able to run it with the VMware Workstation 12 Pro and also with Virtualbox.

My suggestions would be
1- Try following this tutorial using the provided .vmdk file: https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=2010196

2- Alternatively, try using VirtualBox instead

Let me know if any of these worked for you

**Comment by hborhan** [comment]
Thanks for your reply. I could run it with the VirtualBox as you
recommended. After I start the virtual machine, there is a feedback control
account option and a guest session option. I could log into the guest
session as I do not have the feedback control session password. However in
the guest session, there is not any matlab installed. Then I tried to
download and install Matlab from Purdue university website. I am teaching
at Purdue. However when i try to install it using :sudo command, it says I
cannot change the root. It seems I need to have admin access to do that.
Would you know what I need to do to have access to Matlab on the virtual
machine? I am not familiar with Linux much and I appreciate your help on
this.
Thanks,
Ali

**Comment by friether** [comment]
I haven't been working with the toolbox for a while now,  but in that directory where you downloaded the vmdk there should be a readme with some information

**Comment by friether** [comment]
The gettingstarted.pdf instructs how to prepare a Ubuntu from scratch, that's always an option 
Maybe guivenca has additional comments

**Comment by nbibin001** [comment]
All I was able to get the virtual machine setup using the VMWare Workstation 12 Pro. And installed Matlab 2016b. Then was trying to follow the steps mentioned in the MIT_MatlabToolbox/media/GettingStarted.pdf. So first I saved the Drone MAC address, and then copied the rollingspider.edu.plf file using the EDUfirmwareUploadSYS script. Then inserted the battery to the drone after removing the USB connection from the computer, the LED's blinked alternately, after they became solid I tried to connect to the drone using Bluetooth using the DroneConnect.sh file. But it says it cant connect to the MAC address. I tried several times but no luck. I am attaching a scheenshot here. The bluetooth is enabled in the computer. Could anyone please suggest if I am missing something here ? Appreciate your help.

![ubuntu_screen](https://cloud.githubusercontent.com/assets/15659590/19095608/aaa86426-8a64-11e6-8e41-5b8f9e3d4770.png)

**Comment by nbibin001** [comment]
In continuation to the above, I tried to manually pair the drone. So it did it show up and I could pair in the virtual machine as well. Still the script DroneConnect.sh was not able to connect. Still displays the same error as shown above.
![bluetooth_scanning](https://cloud.githubusercontent.com/assets/15659590/19111673/d98ba036-8ace-11e6-8e63-0d57636f362f.PNG)
![bluetooth_device_paired](https://cloud.githubusercontent.com/assets/15659590/19111676/ddd8d172-8ace-11e6-8492-37bf6f91d73f.PNG)

**Comment by guivenca** [comment]
@nbibin001 I had a similar issue and I think the problem is in the name convention used for connections. If you type `ifconfig`, you will see a list of network interfaces. Most likely `eth0` and `lo` will be included and there will probably be at least another one associated with the bluetooth. It may be something `bnep0`, `br0`, or even enp#s# ( some numbers). Could you try running that command and identifying  the proper name (let's call it `link` )?

If it's different from `bnep0`, go to MIT_MatlabToolbox/bin/ and open the file DroneConnect.sh. Near the end you will see a line `ifconfig bnep0 192.168.1.2 up` ( and some other stuff). Try changing `bnep` to `link`, the name you identified earlier. Let me know if that works for you ( worked for me).

@hborhan Were you able to find the readme from where you downloaded the VM? It should have the password for you to run any sudo command

**Comment by nbibin001** [comment]
Hello @guivenca , thanks a lot for the suggestions. I was trying the ifconfig on the virtual machine. It only shows two interfaces, eth0 and lo. The Bluetooth interface is not showing in here. Am I missing something ? Thanks for your help. Really appreciate it.
![ifconfig_screen](https://cloud.githubusercontent.com/assets/15659590/19187901/3645b184-8c5c-11e6-9cf1-67eba83cea78.PNG)

**Comment by nbibin001** [comment]
I used the "hcitool scan " command to see the Ubuntu system is detecting the drone. It actually does as shown in the screenshot below. And the I tried the command "l2ping" to connect to it using the MAC address and it shows the error cant connect operation now in progress. This is the same error that I am getting when running the DroneConnect.sh as well. Could you please tell if I missing something here ? I was actually able to connect to the drone with my Phone Free Flight and App and was able to control it earlier. So when it showed the message operation now in progress I thought may be it still is connected to phone, so I turned off my bluetooth on phone and tried again, but still no luck. Appreciate any help with this. Thank you.

![bluetooth_scanning_and_connect_command](https://cloud.githubusercontent.com/assets/15659590/19806282/e549d836-9ce7-11e6-8e58-34e339e3e6be.PNG)

**Comment by nbibin001** [comment]
All, I just wanted to update that the Bluetooth connection issue was resolved when I installed the Ubuntu on my computer and stopped using the Virtual machine suggested in the user guide. There seems to be some problem with connecting to the Drone over Bluetooth from the Virtual Machine. It always says "Operation now in Progress". After I installed the Obuntu , it just connected fine in the first attempt itself.
So I moved further along, installed Matlab , simulated the controller, generated code and was using the script to compile and upload the code to the drone using DroneUploadEmbeddedCode.sh. But this time it gives me an error saying that 
"make : /opt/arm-2012.03/bin/arm-none-linux-gnueabl-gcc: command not found" . I guess this is something to do with the compiler path. Not sure what exactly am I missing here yet. Below is the screenshot.
![drone_compiler_error](https://cloud.githubusercontent.com/assets/15659590/19864671/96f5a3e2-9f6f-11e6-8411-f38b3501e822.jpg)

Basically I followed the below steps and extracted the folder [ROSMAT]/libs/gcc-arm-Toolchain to /opt/arm-2012.03/libexec. I hope that is what we are supposed to do, do we need to do anything else ? Could you please help ? 
![drone_compiler_error_2](https://cloud.githubusercontent.com/assets/15659590/19864753/fb78dab4-9f6f-11e6-949a-c77c15d6e830.jpg)

**Comment by friether** [comment]
Check out the latest major evolution of this toolbox https://www.mathworks.com/hardware-support/parrot-minidrones.html
